packages/cube/pipeline/cube-validation.ttl

```diff
@@ -0,0 +1,229 @@
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix code: <https://code.described.at/> .
+@prefix p: <https://pipeline.described.at/> .
+@prefix sparql: <https://barnard59.zazuko.com/operations/sparql/> .
+@prefix http: <https://barnard59.zazuko.com/operations/http/> .
+@prefix shacl: <https://barnard59.zazuko.com/operations/shacl/> .
+@prefix base: <https://barnard59.zazuko.com/operations/base/> .
+@prefix n3: <https://barnard59.zazuko.com/operations/formats/n3/> .
+@prefix ntriples: <https://barnard59.zazuko.com/operations/formats/ntriples/> .
+
+@base <http://barnard59.zazuko.com/pipeline/cube-validation/> .
+
+_:endpoint a p:Variable ;
+  p:name "endpoint" ;
+  rdfs:label "SPARQL endpoint" ;
+  p:value "https://lindas.admin.ch/query"
+.
+
+_:cube a p:Variable ;
+  p:name "cube" ;
+  rdfs:label "cube URI" ;
+.
+
+_:constraint a p:Variable ;
+  p:name "constraint" ;
+  rdfs:label "cube constraint file" ;
+.
+
+_:profile a p:Variable ;
+  p:name "profile" ;
+  rdfs:label "cube constraint profile URL" ;
+  p:value "https://cube.link/latest/shape/standalone-constraint-constraint" ;
+.
+
+_:batchSize a p:Variable ;
+  p:name "batchSize" ;
+  rdfs:label "batch size" ;
+  p:value 50 ;
+.
+
+_:maxViolations a p:Variable ;
+  p:name "maxViolations" ;
+  rdfs:label "max number of violations" ;
+  p:value 3 ;
+.
+
+
+<fetch-cube-constraint> a p:Pipeline ;
+  p:variables [ p:variable _:endpoint, _:cube ] ;
+  p:steps
+    [
+      p:stepList
+        (
+          _:queryConstraint
+          [ ntriples:serialize () ]
```

Comment:
Actually, is it even necessary to parse and then serialize again? Looks to me that you'd be simply able to stream the query results if you request n-triples

Reply:
that would mean to use a generic http fetch instead of the barnard59-sparql operations? Because my understanding is that the 'construct' operation returns a stream of quads

Reply:
https://github.com/zazuko/barnard59/issues/206